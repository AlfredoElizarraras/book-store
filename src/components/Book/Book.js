import React from 'react';
import PropTypes from 'prop-types';
import CircleProgressChart from '../CircleProgressChart';
import Button from '../Button';
import '../../flex.css';
import './index.css';

const Book = ({ book, handleRemoveBook }) => (
  <article id={book.id} className="flex book jc-sb">
    <div>
      <div>
        <span className="book__main-info__category">{book.category}</span>
        <h2 className="book__main-info__title">{book.title}</h2>
        <span className="book__main-info__author">{book.author}</span>
      </div>
      <div className="book__action-links">
        <a href="#the-hunger-games" className="book__action-link">
          Comments
        </a>
        <a
          href="#the-hunger-games"
          className="book__action-link"
          onClick={() => handleRemoveBook(book.id)}
        >
          Remove
        </a>
        <a
          href="#the-hunger-games"
          className="book__action-link book__action-link--last"
        >
          Edit
        </a>
      </div>
    </div>
    <div className="flex jc-sb">
      <div className="flex jc-c ai-c book__chart-wrapper">
        <CircleProgressChart
          percent={book.progress}
          radius="35"
          strokeWidth="5"
          maxValue={book.maxProgressValue}
          currentValue={book.currentProgressValue}
        />
        <span className="book__chart-wrapper--border-right" />
      </div>
      <div className="flex flex-col">
        <span className="book__progress-measure">
          Current
          {' '}
          {book.progressMeasure}
        </span>
        <span className="book__current-progress-value">
          {book.progressMeasureValue}
        </span>
        <Button margin="0.6rem 0px 0px">Update progress</Button>
      </div>
    </div>
  </article>
);

Book.propTypes = {
  book: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    progress: PropTypes.number,
    author: PropTypes.string,
    progressMeasure: PropTypes.string,
    progressMeasureValue: PropTypes.string,
    maxProgressValue: PropTypes.number,
    currentProgressValue: PropTypes.number,
  }),
  handleRemoveBook: PropTypes.func.isRequired,
};

Book.defaultProps = {
  book: {
    progress: 30,
    author: 'Book author',
    progressMeasure: 'chapter',
    progressMeasureValue: 'Introduction',
    maxProgressValue: 0,
    currentProgressValue: 0,
  },
};

export default Book;
