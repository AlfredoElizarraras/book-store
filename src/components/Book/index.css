@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@700&display=swap');

:root {
  --black-three: #121212;
}

.book {
  margin-bottom: 1rem;
  padding: 1.5rem 3rem 1.5rem 1rem;
  border: 1px solid #e8e8e8;
  background-color: #fff;
}

.book__main-info__title {
  color: var(--black-three);
  font-size: 1.375rem;
  font-weight: bold;
  letter-spacing: -0.2px;
  margin-top: 0.25rem;
}

.book__main-info__category {
  opacity: 0.5;
  font-family: 'Monserrat', sans-serif;
  font-size: 0.875rem;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: normal;
  color: var(--black-three);
}

.book__main-info__author {
  font-size: 0.875rem;
  font-weight: 300;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #4386bf;
}

.book__action-links {
  margin-top: 1.313rem;
}

.book__action-link {
  text-decoration: none;
  padding-right: 1.0625rem;
  border-right: solid 1px #e8e8e8;
  margin-right: 17px;
  color: #4386bf;
}

.book__action-link--last {
  border: none;
}

.book__chart-wrapper {
  margin-right: 3.6875rem;
}

.book__chart-wrapper--border-right {
  height: 70%;
  background-color: #e8e8e8;
  width: 1px;
  margin-left: 4.938rem;
}

.book__progress-measure {
  margin: 0.063rem 5.375rem 0 0;
  opacity: 0.5;
  font-size: 0.813rem;
  font-weight: 300;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: normal;
  text-transform: uppercase;
  color: var(--black-three);
}

.book__current-progress-value {
  margin: 0.438rem 0 0.25rem 0;
  font-size: 0.865rem;
  font-weight: 300;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: -0.4px;
  color: var(--black-three);
}
