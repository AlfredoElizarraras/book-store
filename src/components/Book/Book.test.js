import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Book from './Book';

describe('Book', () => {
  const props = {
    book: { id: 1, title: 'Book Title', category: 'Book Category' },
  };

  it('renders without crashing', () => {
    const table = document.createElement('table');
    const tbody = document.createElement('tbody');
    table.appendChild(tbody);
    ReactDOM.render(<Book book={props.book} />, tbody);
    ReactDOM.unmountComponentAtNode(tbody);
  });

  test('Has a valid snapshot', () => {
    const component = renderer.create(<Book book={props.book} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
