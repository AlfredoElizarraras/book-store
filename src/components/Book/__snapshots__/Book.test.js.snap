// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Has a valid snapshot 1`] = `
<tr>
  <td>
    1
  </td>
  <td>
    Book Title
  </td>
  <td>
    Book Category
  </td>
</tr>
`;
